File: threads/src/main/java/im/threads/business/database/DBHelper.java
package im.threads.business.database;

import androidx.annotation.NonNull;

import java.util.List;

import im.threads.business.models.ChatItem;
import im.threads.business.models.FileDescription;

public interface DBHelper {

    void cleanDatabase();

    @NonNull
    List<ChatItem> getChatItems(int offset, int limit);

    List<FileDescription> getAllFileDescriptions();
}

File: src/main/java/im/threads/business/database/DBHelper.kt
package im.threads.business.database

import im.threads.business.models.ChatItem
import im.threads.business.models.FileDescription

interface DBHelper {
    fun cleanDatabase()
    fun getChatItems(offset: Int, limit: Int): List<ChatItem?>
    val allFileDescriptions: List<FileDescription?>?
}
